File: windows/dispatch-desktop/Models/AppScreen.cs
// 최상위 메뉴 = 주 화면(§3.4). 관제 캔버스(도킹 호스트)는 항상 마운트돼 있고, 나머지 셋은 그 위에 겹치는 화면 레이어다.
namespace DispatchDesktop.Models;

public enum AppScreen
{
    /// <summary>[관제] — §3.1 도킹 캔버스.</summary>
    Dispatch,
    /// <summary>[이력] — 끝난 세션의 날짜 창 조회 + 녹취 재생(§4.6).</summary>
    History,
    /// <summary>[PTT 그룹] — 관리 범위 안 그룹 목록·생성/편집/삭제(§4.7).</summary>
    PttGroups,
    /// <summary>[관리] — 조직·구성원·번호(§4.5). 관리 범위(dispatch.directoryAdmin)가 있어야 활성.</summary>
    Admin,
}

public static class AppScreens
{
    public static string Title(AppScreen s) => s switch
    {
        AppScreen.History => "이력",
        AppScreen.PttGroups => "PTT 그룹",
        AppScreen.Admin => "조직 · 구성원 · 번호",
        _ => "관제",
    };

    /// <summary>화면 전환 핫키(§8) — 고정 F1~F4. 설정 핫키가 같은 키를 쓰면 설정 쪽이 우선한다.</summary>
    public static AppScreen? OfFunctionKey(System.Windows.Input.Key k) => k switch
    {
        System.Windows.Input.Key.F1 => AppScreen.Dispatch,
        System.Windows.Input.Key.F2 => AppScreen.History,
        System.Windows.Input.Key.F3 => AppScreen.PttGroups,
        System.Windows.Input.Key.F4 => AppScreen.Admin,
        _ => null,
    };
}

File: android/dispatch-tablet/src/main/java/com/cims/ue/dispatch/ui/AppScreen.kt
// 최상위 화면 — 하단 내비 넷 (docs/design/features/android_dispatch_tablet.md §6.2, dispatch_desktop_ui.md §3.4)
//
// 데스크톱의 상단 바 메뉴 4개(F1~F4)를 태블릿에서는 하단 내비로 낸다. 별창은 없고 같은 창의
// 레이어 전환이며, 화면 상태는 앱 수명 동안 유지한다(폼 입력이 탭 전환으로 날아가지 않게).
package com.cims.ue.dispatch.ui

/** 최상위 화면. 순서가 곧 하단 내비의 배열이다. */
enum class AppScreen(val label: String, val hotkey: String) {
    DISPATCH("관제", "F1"),
    HISTORY("이력", "F2"),
    PTT_GROUPS("PTT 그룹", "F3"),
    ADMIN("관리", "F4");

    /** 관제 밖 화면에는 상단에 관제 요약 띠가 붙는다(§6.2). */
    val showsSummaryStrip: Boolean get() = this != DISPATCH

    companion object {
        /** 하드 키보드가 있으면 F1~F4 도 받는다(§7). */
        fun ofFunctionKey(keyCode: Int): AppScreen? = when (keyCode) {
            android.view.KeyEvent.KEYCODE_F1 -> DISPATCH
            android.view.KeyEvent.KEYCODE_F2 -> HISTORY
            android.view.KeyEvent.KEYCODE_F3 -> PTT_GROUPS
            android.view.KeyEvent.KEYCODE_F4 -> ADMIN
            else -> null
        }
    }
}

/** 관제 캔버스의 탭 둘 — 6패널을 접은 결과(§6.3). */
enum class DispatchTab(val label: String) {
    PTT("PTT"),
    CALLS("일반통화"),
}
